docs/kepler_model_server/pipeline.md

```diff
@@ -1,25 +1,54 @@
 
-**AbsPower:** the predicted power is a node power that includes nearly-static power at idling state and dynamic power when running the process on the machine.
+From Kepler queries, the default extractor generates a dataframe with the following columns.
+
+timestamp|features|labels[unit]\_[#unit]\_[component]\_power|node type
+---|---|---|---
+timestamp|e.g.,<br>cgroupfs_cpu_usage_us<br>cgroupfs_memory_usage_bytes<br>cgroupfs_system_cpu_usage_us<br>cgroupfs_user_cpu_usage_us|e.g.,<br>package_0_package_power<br>package_1_package_power<br>package_0_core_power<br>package_1_core_power<br>package_0_uncore_power<br>package_1_uncore_power<br>package_0_dram_power<br>package_1_dram_power|node_type
+
+
+![](../fig/model-server-e2e.png)
+
+<!-- TOC tocDepth:2..3 chapterDepth:2..6 -->
+
+- [Labeling energy source](#labeling-energy-source)
+- [Idle power/Control plane power](#idle-powercontrol-plane-power)
+- [Available metrics](#available-metrics)
+- [Learning methods](#learning-methods)
+- [Node Spec](#node-spec)
+
+<!-- /TOC -->
 
-**DynPower:** the predicted power is per-process/per-container power that excludes nearly-static power at idling state.
+## Labeling energy source
 
-### Power source
+`energy source` or `source` refers to the source (power meter) that provides an energy number. Each source provides one or more `energy components`. Currently supported source are shown as below.
 
-Power source|Power components
+Energy/power source|Energy/power components
 ---|---
 [rapl](../design/kepler-energy-sources.md#rapl---running-average-power-limit)|package, core, uncore, dram
 [acpi](../design/kepler-energy-sources.md#using-kernel-driver-xgene-hwmon)|platform
+||
 
-## Model Features
-The features are groupped by the metric sources.
+## Idle power/Control plane power
 
-### Feature group
+`isolate` is a mechanism to separate the power portion that is consumed on the node in an idle state or the power portion that is consumed by operating systems and control plane processes. These portions of power is more than zero even if the metric utilization of workload is zero. We called the models those are trained after isolating these power portions as `DynPower` models. At the same time, the models those are trained without isolation are called `AbsPower` models. `DynPower` model is used to estimate `container power` and `process power` while `AbsPower` model is used to estimate `node power`.
+
+There are two common available `isolators`: *ProfileIsolator* and *MinIdleIsolator*. 
+
+*ProfileIsolator* relies on profiled background powers (profiles) and removes resource usages by system processes from the training while *MinIdleIsolator* assumes minimum power as an idle power and includes resource usages by system processes in the training. 
+
+The pipeline with *ProfileIsolator* will be applied first if the profile that matches the training `node_type` is available. Otherwise, the other pipeline will be applied. 
```

Comment:
> The pipeline with *ProfileIsolator* will be applied first if the profile that matches the training `node_type` is available.

What does it mean? What is the `node_type`? When it does not match?

> Otherwise, the other pipeline will be applied.

Can you explain it better?

> *ProfileIsolator* relies on profiled background powers (profiles) and removes resource usages by system processes from the training

If we don't know the power consumption of the system processes before hand, how can we remove the resource usages of the system processes?

Reply:
It took the Kepler metric in idle state. The containers that run in idle state is considered as system processes/control plane.